### PR TITLE
Implemented LXC (Linux Containers) management support

### PR DESCRIPTION
**Pull request by litwol**
### Sample Rexfile -- Adjust container name according to your environment.
use Rex -feature => ['1.4'];

use Data::Dumper;

use Rex::Commands::Virtualization;
set virtualization => "Lxc";

task "dump_info", sub {
  dump_system_information;
};

task "lxc_list", sub {
  my @out = vm list => "active",
    fancy => 1;

  print Dumper \@out;
};

task "lxc_info", sub {
  my @out = vm info => "rexify";

  print Dumper \@out;
};

task "lxc_start", sub {
  my @out = vm start => "gentoo-base";

  print Dumper \@out;
};

task "lxc_stop", sub {
  my @out = vm stop => "gentoo-base",
  timeout => 10;

  print Dumper \@out;
};

task "lxc_create", sub {
  my @out = vm create => "vitae_dolores",
    template => "gentoo";

  print Dumper \@out;
};

task "lxc_copy", sub {
  my $out = vm copy => "some-base",
    newname => "vitae_dolores",
    ephemeral => TRUE,
    mount => 'bind=/data/brand/home/devadmin:home/devadmin,bind=/data/brand/var/lib/mysql:/home/mysql_test:ro' ;

  my @out = vm info => $out;

  print Dumper \@out;
};

task "lxc_destroy", sub {
  my @out = vm destroy => "vitae_dolores",
    snapshots => 1;

  print Dumper \@out;
};

### CODE REVIEW

File: lib/Rex/Virtualization/Lxc.pm [hunk 1]
```diff
@@ -0,0 +1,49 @@
+#
+# (c) Oleg Hardt <litwol@litwol.com>
```
Comment (litwol) [inline lib/Rex/Virtualization/Lxc.pm]: Please some one advise whether my name supposed to go here, or "(c) Jan Gehring <jan.gehring@gmail.com>"... i'm really not sure. 

thanks.

File: lib/Rex/Virtualization/Lxc.pm [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+ fancy => 1;
+
+=cut
+
+package Rex::Virtualization::Lxc;
+
+use strict;
+use warnings;
+
+our $VERSION = '1.4.0'; # VERSION
```
Comment (krimdomu) [inline lib/Rex/Virtualization/Lxc.pm]: @litwol you don't need to write `our $VERSION = ...` there. Just add the comment `# VERSION`
The release process then will add the right version there.

File: lib/Rex/Virtualization/Lxc.pm (hunk 1 above)
Comment (ferki) [inline lib/Rex/Virtualization/Lxc.pm]: Since this is code you wrote, you own copyright over it, so it's best ot indicate it as such :wink:

**Comment by ferki** [comment]
@litwol: please take a look at the failing tests (indicated by clickable the red X next to each commit, and at the test summary at the bottom of the PR), and fix them. It seems as simple as applying the project's perltidy rules to the files, and fixing a minimum perl requirement issue.

Other than that, based on my first quick read-through it looks nice :+1:

**Comment by krimdomu** [comment]
@litwol 

thanks for your work. I've just squashed your commits and merged it into development branch. And i'm also going to merge it into development-1.x branch.

**Comment by litwol** [comment]
Thank you!

**Pull request closed without merging**